Sources/SwiftInspectorCommands/Tests/PropertyAnalyzerCommandSpec.swift

```diff
@@ -0,0 +1,106 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Nimble
+import Quick
+import Foundation
+
+@testable import SwiftInspectorCore
+
+final class PropertyAnalyzerCommandSpec: QuickSpec {
+  override func spec() {
+    describe("run") {
+
+      context("with no arguments") {
+        it("fails") {
+          let result = try? TestTask.run(withArguments: ["properties"])
+          expect(result?.didFail) == true
+        }
+      }
+
+      context("when path is invalid") {
+        it("fails when empty") {
+          let result = try? TestTask.run(withArguments: ["properties", "--path", ""])
+          expect(result?.didFail) == true
+        }
+
+        it("fails when not provided") {
+          let result = try? TestTask.run(withArguments: ["properties", "--path"])
+          expect(result?.didFail) == true
+        }
+
+        it("fails when it doesn't exist") {
+          let result = try? TestTask.run(withArguments: ["properties", "--path", "/fake/path"])
+          expect(result?.didFail) == true
+        }
+
+        it("fails when name is present but the file is invalid") {
+          let result = try? TestTask.run(withArguments: ["properties", "--path", "/fake/path", "--name", "SomeType"])
+          expect(result?.didFail) == true
+        }
+      }
+
+      context("when name is passed") {
+        var fileURL: URL!
+
+        beforeEach {
+          fileURL = try? Temporary.makeFile(content: """
+          final class Some {
+            public var thing: String = "Hello, World"
+          }
+          """)
+        }
+
+        afterEach {
+          try? Temporary.removeItem(at: fileURL)
+        }
+
+        it("fails when name is empty") {
+          let result = try? TestTask.run(withArguments: ["properties", "--path", fileURL.path, "--name", ""])
+          expect(result?.didFail) == true
+        }
+
+        it("fails when name is not provided") {
+          let result = try? TestTask.run(withArguments: ["properties", "--path", fileURL.path, "--name"])
+          expect(result?.didFail) == true
+        }
+
+        it("fails when path is empty") {
+          let result = try? TestTask.run(withArguments: ["properties", "--name", "Some"])
+          expect(result?.didFail) == true
+        }
+
+        it("succeeds") {
+          let result = try? TestTask.run(withArguments: ["properties", "--path", fileURL.path, "--name", "Some"])
+          expect(result?.didSucceed) == true
+        }
+
+        it("returns the type name with property information including the name, access, and scope") {
+          let result = try? TestTask.run(withArguments: ["properties", "--path", fileURL.path, "--name", "Some"])
+          expect(result?.outputMessage).to(contain("Some:(thing|public|instance)"))
```

Comment:
I'd suggest outputting something like:

`Some thing public instance` instead. This is a lot easier to parse as output and to provide as arguments for another tool.

I really need to come up with better documentation for these things.